jenkins-design-language/package.json

```diff
@@ -8,16 +8,15 @@
     "storybook": "npm run gulp build && start-storybook -p 9001 --static-dir dist/assets/",
     "build-storybook": "build-storybook --static-dir dist/assets/ --output-dir website/storybook",
     "flow": "flow",
+    "lint": "gulp lint",
     "gulp": "gulp build",
-    "test": "gulp test",
-    "prepublish": "gulp test",
```

Comment:
No prepublish anymore?

Reply:
Absolutely no prepublish. 1. We are not publishing stuff to npm anymore, so it's not useful, 2. prepublish is called anytime that "npm install" is called... so when frontend-maven-plugin invokes npm install, it runs the tests a 2nd time, which adds 1m+ to the build

Reply:
Actually not publishing anymore to npm means that third party integrators cannot use JDL anymore. e.g. it is impossible to test with storybook since the versions are not published anymore. :(

@cliffmeyers ...the whole move to "hpi as only truth" has completely made integration of bo/jdl components in third party components impossible! re-Usability = 0 :(

Reply:
Can you explain what you mean by "third party component" ?  If a Jenkins plugin depends on the JDL and is using `blueocean-maven-plugin` then it will receive the contents of the JDL's "dist" dir inside of `node_modules/@jenkins-cd/design-language`.

Reply:
Regardless, we can still publish the JDL periodically if something totally disconnected from Blue Ocean wants to use it. `prepublish` isn't really the right hook to be doing a `gulp test` given that `prepublish` runs any time a parameterless `npm install` is run.

Reply:
We could start by publishing JDL and core-js whenever we do a minor or major release of Blue Ocean, e.g. 1.4 or 2.0. Would that help?

Reply:
IMO **could** not but **have to**, for third party is e.g. storybook a good example. You do only get the "correct" version **after** you have installed/started a jenkins, which IMO defeats the purpose of storybook. My point in using storybook to not have to start jenkins. Storybook should get their deps via a **normal** `npm i`.

Reply:
You really just need to run a Maven target to get the right deps, such as `mvn process-sources`. I realize it's not ideal, but you could built that into npm scripts or write a shell script to it for you.

Reply:
So the verdict is `jdl` is a jenkins only component where you need full java installed to be able to use it.

...there is *no* need for that but I guess to do `npm publish` is just too much.

Reply:
No as stated elsewhere it could and probably should be published to npm cc @kzantow could we still publish it? (Separate PR)

Reply:
Right @scherler we can still publish to npm, there's nothing preventing that. The only possible issue is versions causing some horrible annoyances with the `blueocean-maven-plugin`, but I could probably change that to write an appropriate version.